File: Codes/Hash/Non-Cryptographic/RSHash/RSHash.py
# Hash function by Robert Sedgwicks
# Archive of Reversing.ID
# Non-Cryptographic Hash

def RSHash(key):
    a     = 378551
    b     =  63689
    state =      0

    for i in range(len(key)):
        state = state * a + ord(key[i])
        a = a * b
    
    return state

File: Codes/Hash/Non-Cryptographic/RSHash/RSHash.rb
# Hash function by Robert Sedgwicks
# Archive of Reversing.ID
# Non-Cryptographic Hash

def rshash( key, len=key.length )
    a     = 63689
    b     = 378551
    state = 0

    len.times{ |i|
        state = state * a + key[i]
        a *= b
    }
    return state
end